dev/tools/controllerbuilder/pkg/commands/generatetypes/generatetypescommand.go

```diff
@@ -122,23 +126,26 @@ func RunGenerateCRD(ctx context.Context, o *GenerateCRDOptions) error {
 		return err
 	}
 
-	if o.ResourceProtoName != "" {
-		kind := o.ResourceKindName
-		if kind == "" {
-			output := []rune{unicode.ToUpper(rune(o.ResourceProtoName[0]))}
-			for _, val := range o.ResourceProtoName[1:] {
-				output = append(output, val)
-			}
-			kind = string(output)
-		}
-		if !scaffolder.TypeFileNotExist(kind) {
-			fmt.Printf("file %s already exists, skipping\n", scaffolder.GetTypeFile(kind))
-		} else {
-			err := scaffolder.AddTypeFile(kind, o.ResourceProtoName)
-			if err != nil {
-				return fmt.Errorf("add type file %s: %w", scaffolder.GetTypeFile(kind), err)
-			}
+	kind := o.ResourceKindName
+	if kind == "" {
+		kind = o.ResourceProtoName
+	}
```

Comment:
Actually my experience is that the Kind and resourceProtoName are always different, because Kind always has the service name prefixed, while resourceProtoName is always ignore that (which makes sense for both case due to the different conditions). So I think may be we shall make both --kind and --proto-resource as required to stabilize the output.

Reply:
Yeah makes sense. Updated